Expensify/expensify-common
# Update regex of italic

### Details
When entering ___ (3 underscores), the italic characters (underscores) got parsed and only leave one underscore after parsed.
It has 3 changes compared to the current italic regex we're using:

- Replace `(?!\s)` by `(?![\s_])`, so that it doesn't match _ at the beginning of the content

- Replace` \S` by `[^\s_]`, so that it doesn't match _ at the end of the content

- Replace the word boundary \b by a regex at the beginning and end `([^\W_]?)` and `(?![^\W_])` so that the word boundary will also exclude the _ as word, so that cases like __content_ and _content__ will work correctly.

Also we need to update this replacement so that it will not replace if the match starts with a word character (this logic works with `([^\W_]?))`

### Fixed Issues

$ https://github.com/Expensify/App/issues/17665
PROPOSAL: https://github.com/Expensify/App/issues/17665#issuecomment-1516037067


### Tests

To test, we need to update the hash commit of expensify-common package in App repo into `6a781aa8f027d87f78e5dd808759795631fc30e5` like this:
    `"expensify-common": "git+ssh://git@github.com/Expensify/expensifycommon.git#6a781aa8f027d87f78e5dd808759795631fc30e5"`

Open any chat and send these messages and verify that
1. `___`                                   --> Verify that this message isn't parsed
2. `_italic__`                           --> Verify that this message is parsed
3. `__italic_`                           --> Verify that this message is parsed
4. `_italic_another_italic_`    --> Verify that this message is parsed
5. `_italic_`                             --> Verify that this message is parsed
6. `s_italic`                             --> Verify that this message isn't parsed
7. `italic_s`                             --> Verify that this message isn't parsed
8. 
```
_italic

italic
italic_
```   
--> Verify that this message is parsed 
9. 
```
_
_
_
```  
--> Verify that this message isn't parsed
10. 
```
_
_test
_
```                        
--> Verify that this message isn't parsed




### Offline tests


### QA Steps

1. Open any chat
2. Send these messages and verify that
`___`                                   --> Verify that this message isn't parsed
`_italic__`                           --> Verify that this message is parsed
`__italic_`                           --> Verify that this message is parsed
`_italic_another_italic_`    --> Verify that this message is parsed
`_italic_`                             --> Verify that this message is parsed
`s_italic`                             --> Verify that this message isn't parsed
`italic_s`                             --> Verify that this message isn't parsed


### Screenshots/Videos
<details>
<summary>Web</summary>

<img width="1440" alt="Screenshot 2023-05-03 at 16 32 56" src="https://user-images.githubusercontent.com/129500732/235883892-d3bf2100-8390-48bc-9769-7a6f41e02bae.png">





</details>

<details>
<summary>Mobile Web - Chrome</summary>

<img width="309" alt="Screenshot 2023-05-03 at 16 38 15" src="https://user-images.githubusercontent.com/129500732/235883906-c17d29cd-a2c0-45e5-9a7e-50822ef18b12.png">



</details>

<details>
<summary>Mobile Web - Safari</summary>


<img width="288" alt="Screenshot 2023-05-03 at 16 38 36" src="https://user-images.githubusercontent.com/129500732/235883922-0b1f55e8-d79e-40a4-afac-b79662645d8d.png">


</details>

<details>
<summary>Desktop</summary>
<img width="1207" alt="Screenshot 2023-05-03 at 16 42 28" src="https://user-images.githubusercontent.com/129500732/235883993-bed99233-2e70-4ad7-8aae-c866595243b6.png">




</details>

<details>
<summary>iOS</summary>

<img width="405" alt="Screenshot 2023-05-03 at 16 59 57" src="https://user-images.githubusercontent.com/129500732/235888960-0fee6b82-baa2-4c3c-bf68-f562609c935a.png">



</details>

<details>
<summary>Android</summary>

<img width="355" alt="Screenshot 2023-05-03 at 17 11 05" src="https://user-images.githubusercontent.com/129500732/235888978-d455e6a5-9e6f-48dd-9caa-12ecb6a190b4.png">



</details>
